AVADA.pm

```diff
@@ -0,0 +1,244 @@
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=head1 CONTACT
+
+ Ensembl <http://www.ensembl.org/info/about/contact/index.html>
+
+=cut
+
+=head1 NAME
+
+ AVADA
+
+=head1 SYNOPSIS
+./vep -i variations.vcf --plugin AVADA,file=path/to/file
+./vep -i variations.vcf --plugin AVADA,file=path/to/file,feature_match_by=<gene_symbol|ensembl_gene_id>
+
+=head1 DESCRIPTION
+
+Automatic VAriant evidence DAtabase is a novel machine learning tool that uses natural language processing 
+to automatically identify pathogenic genetic variant evidence in full-text primary literature about 
+monogenic disease and convert it to genomic coordinates.
+
+Please cite the AVADA publication alongside the VEP if you use this resource:
+https://pubmed.ncbi.nlm.nih.gov/31467448/
+
+NB: The plugin currently does not annotate for downstream_gene_variant and upstream_gene_variant 
+
+Pre-requisites
+1) AVADA data is available for GRCh37 and can be downloaded from: 
+http://bejerano.stanford.edu/AVADA/avada_v1.00_2016.vcf.gz
+
+wget http://bejerano.stanford.edu/AVADA/avada_v1.00_2016.vcf.gz
+
+2) The file needs to be tabix indexed. You can do this by following commands:
+
+gzip -d avada_v1.00_2016.vcf.gz
+bgzip avada_v1.00_2016.vcf 
+tabix avada_v1.00_2016.vcf.gz
+
+3) As you have already noticed, tabix utility must be installed in your path to use this plugin. 
+
+Options are passed to the plugin as key=value pairs:
+
+The plugin can then be run to retrieve AVADA annotations. 
+By default, the variants are matched with the HGNC gene symbol
+./vep -i variations.vcf --plugin AVADA,file=path/to/file
+
+The output always includes the following columns:
+AVADA_GENE_ID
+AVADA_PMID or AVADA_PMID_WITH_VARIANT_STRING
+
+AVADA_GENE_ID: Gene ID associated with variant as reported by AVADA
+AVADA_PMID: PubMed ID evidence for the variant as reported by AVADA
+AVADA_PMID_WITH_VARIANT_STRING: PubMed ID evidence for the variant as reported by AVADA along with the RefSeq id and the original variant string
+
+The plugin can optionally be run by specifying the feature to match with
+
+In order to match by HGNC gene symbol:
+./vep -i variations.vcf --plugin AVADA,file=path/to/file,feature_match_by=gene_symbol 
+
+In order to match by ENSEMBL gene identifier :
+./vep -i variations.vcf --plugin AVADA,file=path/to/file,feature_match_by=ensembl_gene_id
+
+The plugin can also be run to report the original variant string associated with a RefSeq trancript/protein id 
+reported in the publication. The RefSeq identifier from VEP cache/database is matched with AVADA data.
+./vep -i variations.vcf --plugin AVADA,file=path/to/file,original_variant_string=1
```

Comment:
Can the variant string be reported without transcript matching?
This is useful if you are annotating with Ensembl transcripts as you can match the transcript or protein level change.

This paper: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4303590/ talks about ISG15: c.163C>T rather than a particular transcript
It would be useful to look at the paper when you are annotating against Ensembl transcripts and see ENST00000649529.1:c.163C>T

Reply:
Hi @sarahhunt , I added an additional option to match by RefSeq transcript/protein allowing variant string to also be reported without transcript matching. Please let me know if this makes sense. Thank you

Reply:
Thanks - it's useful to be able to get the variant string without filtering on feature.
 
It would be simpler to have the RefSeq match as an option on feature_match_by